python/ray/rllib/optimizers/aso_aggregator.py

```diff
@@ -105,8 +105,13 @@ def iter_train_batches(self, max_yield=999):
             self.batch_buffer.append(sample_batch)
             if sum(b.count
                    for b in self.batch_buffer) >= self.train_batch_size:
-                train_batch = self.batch_buffer[0].concat_samples(
-                    self.batch_buffer)
+                if len(self.batch_buffer) == 1:
+                    # make a defensive copy to avoid sharing plasma memory
+                    # across multiple threads
```

Comment:
Do we understand what the issue is? The plasma client should be thread safe now after https://github.com/apache/arrow/pull/4503

Reply:
Hmm, let me try it again. It's possible I wasn't using the latest wheels.

Reply:
Oh I know why, it's because I was not able to test with versions of Ray after https://github.com/ray-project/ray/issues/5125

That almost certainly excludes arrow versions with that patch.